enhancements/oc/day2-add-nodes.md

```diff
@@ -0,0 +1,269 @@
+       ipv4:
+         enabled: true
+         address:
+           - ip: 192.168.122.3
+             prefix-length: 23
+         dhcp: false
+```
+
+### monitor-add-nodes command
+
+The format of the command will be the following:
+
+`oc -–kubeconfig=’’ adm monitor-add-nodes <IPs list>`
+
+Where:
+- `–kubeconfig=’’``. Path to the kubeconfig file (or KUBECONFIG if available)
+- `IPs list`` is a list of the nodes IP to be monitored
+
+The command will try to connect to each specified node to retrieve initially the results of the pre-flight validations, and then 
+the progress of the activity. In case of validation failures, the command will report them to the console to allow the user to 
+take the requested actions and troubleshoot the issues. In the future this command could be enhanced to automatically approve the node CSRs
+(identifying the correct CSR requires the Node IP address, which this command already has as an input).
+
+### Workflow Description
+The working environment must satisfy the initial following conditions:
+- `oc` tool is available
+- A direct connection to the target cluster is available
+- kubeconfig file is available
+
+The following steps illustrates the expected workflow:
+1. The cluster administrator prepares the `nodes-config.yaml` file listing all the nodes to be added, optionally with their initial network configuration
+2. The cluster administrator runs the `oc adm add-nodes-image` command to generate the ISO
+3. The cluster administrator boots every node previously listed using the freshly generated ISO
+4. The cluster administrator runs the `oc adm monitor-add-nodes` command to monitor the activity progress
+5. Once a node csr is pending, the cluster administrator runs the `oc adm certificate approve <csr-id>` to complete the process for the specified node
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints
+The implementation will largely reuse the same approach / code present in the openshift/installer repo for creating an ISO image 
+for the installation, with a number of exceptions that will be detailed in the subsequent paragraphs.
+
+#### Proposed architecture
+Every node booted with the generated ISO will be independent for adding itself to the cluster - without any need to specify a single rendezvous IP, 
+as it happens for the installation scenario. In this way if a node is failing for some reason (for example, the pre-flight validations block the process), 
+the other nodes may continue the joining process undisturbed.
+The following hilights better the proposed architecture:
+
+![alt Architecture diagram](day2-add-nodes-diagram.png)
+
+#### Assisted-service interaction
+The following API calls will be used to add a single node to the target cluster:
+- `/api/assisted-install/v2/clusters/import` to import an existing cluster
+- `/api/assisted-install/infra-envs/${INFRA_ENV_ID}/hosts/${HOST_ID}` to add the specified host to the cluster
+
+#### oc integration notes
```

Comment:
I wonder if there's a way to structure things so that this section appears before the detailed description of the `oc` command syntax.

Reply:
I put this part here since it was under the `Implementation Details/Notes/Constraints` section. Maybe we could add a note in the `oc` commands description that the integration details will be provided later in the proposal